File: backend/src/main/java/com/upblit/backend/query/model/Log.java
package com.upblit.backend.query.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "logs")
public class Log {

    @Id
    private String id;

    @Field("clientTimestamp")
    private Instant clientTimestamp;

    @Field("serverTimestamp")
    private Instant serverTimestamp;

    @Field("projectId")
    private Long projectId;

    @Field("applicationId")
    private Long applicationId;

    @Field("trace_id")
    private String traceId;

    @Field("level")
    private String level;

    @Field("type")
    private String type;

    @Field("message")
    private String message;

    @Field("timestamp")
    private Instant timestamp;
}

File: ingestor/models/Log.go
package models

import "time"

type Log struct {
	ClientTimestamp time.Time `json:"clientTimestamp" bson:"clientTimestamp"`
	ServerTimestamp time.Time `json:"serverTimestamp" bson:"serverTimestamp"`

	ProjectID     int64 `json:"projectId" bson:"projectId"`
	ApplicationID int64 `json:"applicationId" bson:"applicationId"`

	TraceID   string    `bson:"trace_id" json:"traceId"`
	Level     string    `bson:"level" json:"level"`
	Type      string    `bson:"type" json:"type"`
	Message   string    `bson:"message" json:"message"`
	Timestamp time.Time `bson:"timestamp" json:"timestamp"`
}
